lib/vrl/compiler/src/compiler.rs

```diff
@@ -10,15 +10,15 @@ use crate::{expression::*, Function, Program, State, Value};
 pub type Errors = Vec<Box<dyn DiagnosticError>>;
 
 pub struct Compiler<'a> {
-    fns: &'a [Box<dyn Function>],
+    fns: &'a [Box<dyn Function + Send + Sync>],
```

Comment:
The VM is stored in a field in the `Remap` transform struct. Implementing `SyncTransform` requires the struct to be `Send + Sync`. 

The VM keeps stores the functions in a Vector. This isn't a strict requirement. An altenative could be to pass the list in when when compiling and again when interpreting. However since the bytecode stores the index in this list of the function to be called rather than the function itself, it could theoretically use one list for compiling and a different list during execution - which would result in hard to track down bugs.